provider/pkg/provider/util/util.go

```diff
@@ -44,3 +52,19 @@ func (w *ConcurrentWriter) Write(bs []byte) (int, error) {
 	defer w.mu.Unlock()
 	return w.Writer.Write(bs)
 }
+
+// TestContext is a test implementation of p.Context that records all log messages in a buffer, regardless of severity.
+type TestContext struct {
+	context.Context
+	Output bytes.Buffer
+}
+
+func (c *TestContext) log(msg string) {
+	c.Output.WriteString(msg)
+}
+
+func (c *TestContext) Log(_ diag.Severity, msg string)                  { c.log(msg) }
+func (c *TestContext) Logf(_ diag.Severity, msg string, _ ...any)       { c.log(msg) }
+func (c *TestContext) LogStatus(_ diag.Severity, msg string)            { c.log(msg) }
+func (c *TestContext) LogStatusf(_ diag.Severity, msg string, _ ...any) { c.log(msg) }
+func (c *TestContext) RuntimeInformation() p.RunInfo                    { return p.RunInfo{} }
```

Comment:
This shouldn't be accessible as part of the non-test code. Can we move it to `util/util_test.go` so it is only usable from tests?